src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -97,8 +111,26 @@ protected override Assembly Load(AssemblyName assemblyName)
         {
             lock (_guard)
             {
+                // First try to satisfy the load from the default context.
+                Assembly assembly;
+                try
+                {
+                    assembly = _defaultContext.LoadFromAssemblyName(assemblyName);
+                    if (assembly != null)
+                    {
+                        return assembly;
+                    }
+                }
+                catch
+                {
+                    // LoadFromAssemblyName indicates failure by throwing exceptions; we need to
+                    // catch them, ignore them, and try to load the assembly ourselves.
+                    // This may mask underlying issues where _defaultContext.LoadFromAssemblyName
+                    // should have succeeded, but there isn't anything we can do about that.
```

Comment:
If the assembly is already loaded, there are two possibilities: it isn't loaded at all, or it is loaded in another load context but not this one.

The exceptions indicate that the default context couldn't locate an assembly with the given name, or located what *should* have been the assembly but wasn't actually (e.g., a native file with the same name, or a corrupted file) or maybe couldn't read the file due to permission issues.

We could think about letting some of these through. For example, if the default context located the assembly but couldn't read it (BadImageFormatException or an exception representing a permission issue) then it would probably be better to let that exception propagate rather than catch or ignore it. It's not clear where we would draw the line, though, and since this whole approach is a workaround for other runtime issues I'd rather just keep it as simple as possible.